x-pack/solutions/observability/packages/kbn-streams-schema/src/models/ingest/fields/index.ts

```diff
@@ -21,17 +21,23 @@ export const FIELD_DEFINITION_TYPES = [
 
 export type FieldDefinitionType = (typeof FIELD_DEFINITION_TYPES)[number];
 
-export interface FieldDefinitionConfig {
+// We redefine type as we only allow a subset of types
+export type FieldDefinitionConfig = MappingProperty & {
   type: FieldDefinitionType;
-  format?: string;
-  additionalProperties?: MappingProperty;
-}
+};
+
+// Parameters that we provide a generic (JSON blob) experience for
+export type FieldDefinitionConfigAdvancedParameters = Omit<
+  FieldDefinitionConfig,
+  'type' | 'format'
+>;
 
-export const fieldDefinitionConfigSchema: z.Schema<FieldDefinitionConfig> = z.object({
-  type: z.enum(FIELD_DEFINITION_TYPES),
-  format: z.optional(NonEmptyString),
-  additionalProperties: z.optional(z.record(z.string(), z.unknown())),
-});
+export const fieldDefinitionConfigSchema: z.Schema<FieldDefinitionConfig> = z.intersection(
+  z.object({
+    type: z.enum(FIELD_DEFINITION_TYPES),
+  }),
```

Comment:
What about `format`? Is we leave it optional, it would change its validation to allow an empty string, which previously was not allowed, shall we add it back among `type`?

Reply:
Hmm, yeah, good point. I missed the explicit `NonEmptyString` there. Ideally what I wanted was only `type` defined explicitly as it diverges from the ES source of truth from `MappingProperty` (as we only support a subset). And then essentially have all the other keys throw from ES if invalid (since the `record` typing would be broad for everything else). I'll add it back though, since it's "first class" it's probably okay.